packages/website/src/components/ErrorsViewer.tsx

```diff
@@ -0,0 +1,115 @@
+import type { ErrorItem } from './types';
+
+import styles from './ErrorsViewer.module.css';
+
+export interface ErrorsViewerProps {
+  readonly value?: ErrorItem[];
+}
+
+export interface ErrorBlockProps {
+  readonly item: ErrorItem;
+  readonly onClick: () => void;
+  readonly isLocked: boolean;
+}
+
+function severityClass(severity: Monaco.MarkerSeverity): string {
+  switch (severity) {
+    case 8:
+      return 'danger';
+    case 4:
+      return 'caution';
+    case 2:
+      return 'note';
+  }
+  return 'info';
+}
+
+function ErrorBlock({ item, onClick, isLocked }: ErrorBlockProps): JSX.Element {
+  return (
+    <div className={`admonition alert alert--${severityClass(item.severity)}`}>
+      <div className="admonition-content">
+        <div className="row row--no-gutters">
+          <div className="col col--12">
+            {item.message} {item.location}
+          </div>
+          {item.hasFixers && (
+            <div className="col col--12">
+              {item.fixers.map((fixer, index) => (
+                <div key={index} className={styles.fixer}>
+                  <span>&gt; {fixer.message}</span>
+                  <button
+                    className="button button--primary button--sm"
+                    disabled={isLocked}
+                    onClick={(): void => {
+                      fixer.fix();
+                      onClick();
+                    }}
+                  >
+                    fix
+                  </button>
+                </div>
+              ))}
+            </div>
+          )}
+        </div>
+      </div>
+    </div>
+  );
+}
+
+export default function ErrorsViewer({
+  value,
+}: ErrorsViewerProps): JSX.Element {
+  const [model, setModel] = useState<[string, ErrorItem[]][]>();
+
+  const [isLocked, setIsLocked] = useState<boolean>(false);
+
+  const onClick = useCallback(() => {
+    setIsLocked(true);
+  }, [isLocked]);
+
+  useEffect(() => {
+    if (value) {
+      setModel(
+        Object.entries(
+          value.reduce<Record<string, ErrorItem[]>>((acc, obj) => {
+            if (!acc[obj.group]) {
+              acc[obj.group] = [];
+            }
+            acc[obj.group].push(obj);
+            return acc;
+          }, {}),
+        ).sort(([a], [b]) => a.localeCompare(b)),
+      );
+    } else {
+      setModel(undefined);
+    }
+    setIsLocked(false);
+  }, [value]);
+
+  if (model) {
+    return (
+      <div className={styles.list}>
+        {model.map(([group, data]) => {
+          return (
+            <div className="margin-top--sm" key={group}>
+              <h4>{group}</h4>
+              {data.map((item, index) => (
+                <ErrorBlock
+                  isLocked={isLocked}
+                  onClick={onClick}
+                  item={item}
+                  key={index}
+                />
+              ))}
+            </div>
+          );
+        })}
+      </div>
+    );
+  }
+
+  return <div className={styles.list}></div>;
```

Comment:
do you need a separate return here?
couldn't this code just be like this, given that it would return `undefined` and render nothing?
```suggestion
  return (
    <div className={styles.list}>
      {model?.map(([group, data]) => {
        return (
          <div className="margin-top--sm" key={group}>
            <h4>{group}</h4>
            {data.map((item, index) => (
              <ErrorBlock
                isLocked={isLocked}
                onClick={onClick}
                item={item}
                key={index}
              />
            ))}
          </div>
        );
      })}
    </div>
  );
```

Reply:
i missed that after refactoring